doschema/merger.py

```diff
@@ -0,0 +1,193 @@
+                elem,
+                self.ignore_index
+            ).traverse(fields_types_dict)
+
+    def process_type_field(self, fields_types_dict, field_value):
+        """Check if a name field with type "type" has already appeared.
+
+        If had appeared with different type,raise an exception.
+        Otherwise, add object to a dict of types.
+
+        :param fields_types_dict: Dictionary keeping path and type of
+                                  the field.
+        :param field_name: Name of the field.
+        :param field_value: Type of the field.
+        """
+        if self.curr_field in fields_types_dict.keys():
+            if fields_types_dict[self.curr_field].field_type != field_value:
+                raise JsonSchemaCompatibilityException({
+                    'field_name': self.curr_field,
+                    'old_type': fields_types_dict[
+                        self.curr_field
+                    ].field_type,
+                    'old_schema': fields_types_dict[
+                        self.curr_field
+                    ].schema_index,
+                    'new_type': field_value,
+                    'new_schema': self.schema_id
+                })
+        else:
+            fields_types_dict[self.curr_field] = FieldToAdd(
+                self.schema_id,
+                self.curr_field,
+                field_value
+            )
+
+    def collect_special(self, fields_types_dict, json_objects_list):
+        """Operate a field with name 'allOf', 'anyOf' or 'oneOf'.
+
+        :param fields_types_dict: Dictionary keeping path and type of
+                                  the field.
+        :param json_objects_list: List of json objects.
+        """
+        for elem in json_objects_list:
+            JsonSchemaCompatibilityChecker(
+                self.schema_id,
+                self.curr_field,
+                elem,
+                self.ignore_index
+            ).traverse(fields_types_dict)
+
+    @staticmethod
+    def starter(schemas_list, ignore_index=True):
+        """Start processing a given list of schemas.
+
+        :param schemas_list: List of schemas to check compatibility.
+        """
+        fields_types_dict = {}
+        ind = 0
+        for scheme in schemas_list:
+            JsonSchemaCompatibilityChecker(
+                ind, (), scheme, ignore_index
+            ).traverse(fields_types_dict)
+            ind += 1
+
+
+class FieldToAdd(object):
+    """Class to keep field description."""
+
+    def __init__(self, schema_index, field_tuple, field_type):
+        """Constructor.
+
+        :param schema_index: Index of schema where field was first declared.
+        :param field_tuple: Tuple with path of a field.
+        :param field_type: Type of a field.
+        """
+        self.schema_index = schema_index
+        self.field_tuple = field_tuple
+        self.field_type = field_type
+
+
+class JsonSchemaCompatibilityException(Exception):
```

Comment:
Right now the main reason for incompatibility is conflicting types. But later there might be other reasons.
Thus I would recommend having a root exception for every compatibility issue, and a child class exception for "type conflicts".
Something similar to what is done in other modules (example in [dojson](https://github.com/inveniosoftware/dojson/blob/master/dojson/errors.py))

Normally I would recommend to have simpler exception classes which just pass their arguments to the `Exception` class, but in this case it can be useful to know which schemas are conflicting so having them as separate attributes is a good idea.